x/ecocredit/server/core/create_project.go

```diff
@@ -35,13 +35,19 @@ func (k Keeper) CreateProject(ctx context.Context, req *core.MsgCreateProject) (
 		return nil, err
 	}
 
-	if err = k.stateStore.ProjectTable().Insert(ctx, &api.Project{
+	project := &api.Project{
 		Id:           projectID,
 		Admin:        adminAddress,
 		ClassKey:     classInfo.Key,
 		Jurisdiction: req.Jurisdiction,
 		Metadata:     req.Metadata,
-	}); err != nil {
+	}
+
+	if req.ReferenceId != "" {
+		project.ReferenceId = req.ReferenceId
+	}
```

Comment:
don't think we actually need this check - if it's empty, the struct will use the default empty string anyway, so the result is this same even if this check is false. we should just set it in the struct above directly